src/libratbag.c

```diff
@@ -1128,6 +1129,11 @@ ratbag_resolution_set_active(struct ratbag_resolution *resolution)
 	struct ratbag_profile *profile = resolution->profile;
 	struct ratbag_resolution *res;
 
+	if (resolution->is_disabled) {
+		log_error(profile->device->ratbag, "%s: setting the active resolution to a disabled resolution is not allowed\n", profile->device->name);
+		return -EINVAL;
```

Comment:
Returning a POSIX error code here and in other modified functions in this file is out of place.
It should return a ratbag_error_code instead.
I guess `RATBAG_ERROR_VALUE` fits best here.